Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm

```diff
@@ -500,20 +523,21 @@ - (id)initWithWindow:(UIWindow *)window
     _sceneMinimumSize = windowScene.sizeRestrictions.minimumSize;
     _sceneChromeOptions = windowScene.mrui_placement.preferredChromeOptions;
     _sceneResizingBehavior = windowScene.mrui_placement.preferredResizingBehavior;
-    _prefersOrnamentsHidden = windowScene.prefersOrnamentsHidden_forLMKOnly;
 
-    _ornamentDepths = [NSMapTable weakToStrongObjectsMapTable];
+    _ornamentProperties = [NSMapTable weakToStrongObjectsMapTable];
 
     MRUIPlatterOrnamentManager *ornamentManager = windowScene._mrui_platterOrnamentManager;
-    for (MRUIPlatterOrnament *ornament in ornamentManager.ornaments)
-        [_ornamentDepths setObject:@(ornament._depthDisplacement) forKey:ornament];
+    for (MRUIPlatterOrnament *ornament in ornamentManager.ornaments) {
+        auto properties = adoptNS([[WKMRUIPlatterOrnamentProperties alloc] initWithDepthDisplacement:ornament._depthDisplacement windowAlpha:ornament.viewController.view.window.alpha]);
```

Comment:
Might be slightly cleaner if the initializer was `initWithOrnament:` and the properties were read in the initializer.